python/lbann/util/class_generator.py

```diff
@@ -96,7 +96,18 @@ def export_proto(self):
             message = getattr(proto, base_field_name)
             message.SetInParent()
         else:
-            proto = getattr(lbann_pb2, message_name)()
+            # TODO (trb 08/01/2019): This list would have to be
+            # updated any time another _pb2 file is created. It might
+            # be better to have this as a global `frozenset`
+            # (ndryden's suggestion) that gets maintained
+            # elsewhere. But this code either works or doesn't get
+            # executed now, so I vote delaying this fix until a need
+            # arises.
+            proto_modules = set([callbacks_pb2, layers_pb2, metrics_pb2, model_pb2, objective_functions_pb2, optimizers_pb2, weights_pb2])
```

Comment:
It may be safer for this to be a list, just to make sure that the loop is deterministic.

Reply:
Why would the loop not be deterministic? `pop()` will eventually kill the set and you'll get an exception.

Reply:
I'm thinking about an edge case where two messages have the same name and different processes store the set entries in a different order.

Reply:
We might want to think of ways to do this using the `full_name` or something to that effect. You're right that naming collisions are a possible issue; a list only ensure that if the wrong one is chosen, it will always be chosen. The better solution would make this order-invariant; let's leave that for a future PR.

Reply:
Just to add: It is almost certainly the case that different processes will store the set entries in different orders. Python sets hash the object, and hashing is non-deterministic across Python invocations.